packages/block-library/src/site-logo/edit.js

```diff
@@ -0,0 +1,283 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { debounce } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { isBlobURL } from '@wordpress/blob';
+import { useState, useEffect } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import {
+	useEntityProp,
+	__experimentalUseEntitySaving,
+} from '@wordpress/core-data';
+import {
+	IconButton,
+	Notice,
+	PanelBody,
+	RangeControl,
+	ResizableBox,
+} from '@wordpress/components';
+import {
+	BlockControls,
+	BlockAlignmentToolbar,
+	BlockIcon,
+	InspectorControls,
+	MediaPlaceholder,
+	MediaReplaceFlow,
+} from '@wordpress/block-editor';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import icon from './icon';
+
+import ImageSize from '../image/image-size';
+
+const ALLOWED_MEDIA_TYPES = [ 'image' ];
+const ACCEPT_MEDIA_STRING = 'image/*';
+const getHandleStates = ( align, isRTL = false ) => {
+	const defaultAlign = isRTL ? 'right' : 'left';
+	const handleStates = {
+		left: {
+			right: true,
+			left: false,
+		},
+		center: {
+			right: true,
+			left: true,
+		},
+		right: {
+			right: false,
+			left: true,
+		},
+	};
+
+	return handleStates[ align ? align : defaultAlign ];
+};
+
+export default function LogoEdit( { attributes: { align, width }, children, className, clientId, setAttributes, isSelected } ) {
+	const [ isEditing, setIsEditing ] = useState( false );
+	const [ url, setUrl ] = useState( null );
+	const [ error, setError ] = useState();
+	const [ logo, setLogo ] = useEntityProp( 'root', 'site', 'sitelogo' );
+	const [ isDirty, , save ] = __experimentalUseEntitySaving(
+		'root',
+		'site',
+		'sitelogo'
+	);
+
+	const mediaItemData = useSelect(
+		( select ) => {
+			const mediaItem = select( 'core' ).getEntityRecord( 'root', 'media', logo );
+			return mediaItem && {
+				url: mediaItem.source_url,
+				alt: mediaItem.alt_text,
+			};
+		}, [ logo ] );
+
+	const { isRTL, isLargeViewport } = useSelect(
+		( select ) => ( {
+			isRTL: select( 'core/block-editor' ).getSettings().isRTL,
+			isLargeViewport: select( 'core/viewport' ).isViewportMatch( 'medium' ),
+		} )
+	);
```

Comment:
Any particular reason for using multiple useSelect calls instead of just one?

Reply:
This question still stands.

Reply:
They are separate because the call to get `mediaItemData` relies on `sitelogo` being loaded from the API, that's why `sitelogo` is passed as a second parameter to `useSelect`.  Is there another way of achieving that with only one `useSelect` call?

Reply:
I believe you can call the first selector, store the result on a local variable (inside mapSelect) and use it if it's available to perform the second call?

Reply:
You're right, very nice! Done in 0d2585b.